### ISSUE TITLE
nullable properties for neo4j entities

### ISSUE THREAD

**Issue by FangmingD** [Impetus/kundera#564]
I have created an entity to be persisted in neo4j through kundera.

But when I don't set all the properties, the find() method will through an exception.

I traced down the error and found out kundera tries to set all the properties from neo4j database, but not all of them exists. (in com.impetus.client.neo4j.GraphEntityMaper getEntityFromNode())

But in JPA, default all the properties are nullable except Id.

**Comment by mevivs** [comment]
Could you please provide more details?  A sample project would be a great help

-Vivek

**Comment by FangmingD** [comment]
I just use the example classes (Actor,Movie,Role) https://github.com/impetus-opensource/Kundera/wiki/Graph-Database-Support.

Then set id of the Actor, Persist it and then find it.
Got NotFoundException in GraphEntityMapper.java line 184.
Because it is trying to get the property of actor name, which was not persisted.

change line 169 to 
PropertyAccessorHelper.set(entity, field, fromNeo4JObject(node.getProperty(columnName,null), field));
will set a default value to null, wont cause exception.

BTW, I'm using Kundera 2.9.1, I checked 2.11, it is not fixed.

Not sure what else it might affect.

**Comment by FangmingD** [comment]
I'll provide a simple code I used.

```
import java.util.HashMap;
import java.util.Map;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.MapKeyJoinColumn;
import javax.persistence.Table;

@Entity
@Table(name = "actor", schema = "oculus")
public class Actor
{
@Id
@Column(name = "ACTOR_ID")
private String id;

@Column(name = "ACTOR_NAME")
private String name;


public String getId() {
    return id;
}

public void setId(String id) {
    this.id = id;
}

public String getName() {
    return name;
}

public void setName(String name) {
    this.name = name;
}
}
```

TEST

```
import java.io.File;

import org.apache.commons.io.FileUtils;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;

import us.oculus.dao.DAOFactory;
import us.oculus.example.entities.Actor;
import us.oculus.kundera.GenericDAO;

public class EntityGeneratorTest {

private static GenericDAO ta;

@Before
public void initDAO() {
    ta = DAOFactory.instance().get("neo4j_test");
}

@Before
public void setup() throws Exception {

     Actor p = new Actor();
     p.setId("Person-1");
     ta.persist(p);
}

@Test
public void should_save_entity() throws Exception {
    assertNotNull(ta.find(Actor.class, "Person-1"));
}

@AfterClass
public static void destroy() throws Exception {
    DAOFactory.instance().closeAll();
    FileUtils.deleteDirectory(new File("/tmp/oculus/test/graph.db"));
}
}
```

Persistence xml

```
<persistence-unit name="neo4j_test">
<provider>com.impetus.kundera.KunderaPersistence</provider>
<class>us.oculus.example.entities.Actor</class>
<properties>
      <property name="kundera.nodes" value="localhost" />
      <property name="kundera.port" value="7474" />
      <property name="kundera.datastore.file.path" value="/tmp/oculus/test/graph.db" />
      <property name="kundera.dialect" value="neo4j" />
      <property name="kundera.client.lookup.class" value="com.impetus.client.neo4j.Neo4JClientFactory" />
      <property name="kundera.transaction.resource.class" value="com.impetus.client.neo4j.Neo4JTransaction" />
    </properties>
</persistence-unit>
```

**Comment by chhavigangwal** [comment]
Fixed in Kundera-2.11.1

**Comment by chhavigangwal** [comment]
Fixed. Released with Kundera-2.12.
